models/stg/sessionization/segment_web_user_stitching.sql

```diff
@@ -1,4 +1,4 @@
-{{config(materialized = 'table')}}
+{{config(materialized = 'view')}}
```

Comment:
If a person computes a certain metric using dbt run --select metric_name, the person will get the metric computed on the last stored upstream table. If materialized as a view, the upstream model which the metric is dependent on will also run, and will give fresh metric data. Also, saves storage space.

Reply:
We only want to persist the tables which matter to the end user, which are metric tables. That's why changed it to a view

Reply:
> If a person computes a certain metric using dbt run --select metric_name, the person will get the metric computed on the last stored upstream table. If materialized as a view, the upstream model which the metric is dependent on will also run, and will give fresh metric data. Also, saves storage space.

Even if this model is being reran in case someone selectively runs a metric built on top of this model, the data in this view would not change since the upstream models that this view is built on are not being refreshed
Moreover, once dbt server is out, we should start materializing all the models that can be used for computing metrics as table to minimize on computation

Reply:
Will be changing to tables